### PR TITLE
plugins/sql: allow datetime functions

### PR DESCRIPTION
**Pull request by JosephGoulden**
Hi. I'm using the SQL plugin to perform analytics on my node. It would be very useful to be able to use the date/time functions so that only the required data is returned.

I've written some queries to test this change and it works as expected.

### CODE REVIEW

File: plugins/sql.c [hunk 1]
```diff
@@ -334,6 +334,26 @@ static int sqlite_authorize(void *dbq_, int code,
...
+			return SQLITE_OK;
+		if (streq(b, "strftime"))
+			return SQLITE_OK;
+		if (streq(b, "time"))
+			return SQLITE_OK;
+		if (streq(b, "timediff"))
+			return SQLITE_OK;
+		if (streq(b, "unixepoch"))
+			return SQLITE_OK;
+		if (streq(b, "current_date"))
```
Comment (niftynei) [inline plugins/sql.c:351]: `current_date` `current_time` and `current_timestamp` are [keywords for SQLite](https://www.sqlite.org/lang_keywords.html), not functions. as far as i can tell via the SQLite documentation, they're only applicable/useful for setting the default value in a CREATE TABLE statement which iiuc isn't useful for running queries. see the default value [section 3.2](https://www.sqlite.org/lang_createtable.html)

Would recommend removing them from this commit.

**Review by niftynei — COMMENTED** [review]
Please remove keyword args from the functions list.

Also remove the unnecessary Merge commit 797cdc82cb084ac2a498d2cfee3c1314ec5ed97b.

File: plugins/sql.c (hunk 1 above)
Comment (JosephGoulden) [reply plugins/sql.c]: Yes you are right, I realised I can use the 'now' argument to the datetime functions to get the current date and time.

**Comment by rustyrussell** [comment]
These look safe, thanks!

Two changes, if you could:
1. Can you add a line to the commit message so this gets mentioned in the CHANGELOG for the release?  Like so:

```
Changelog-Changed: Plugins: `sql` now allows date and time sqlite functions.
```

2. Can you edit `doc/schemas/lightning-sql-template.json` to include these in the list of allowed functions?  That's the basis for our documentation generation.


Thanks!

**Comment by JosephGoulden** [comment]
Do I also need to generate `contrib/msggen/msggen/schema.json` and check that in or is it done elsewhere?

**Comment by rustyrussell** [comment]
> CI failed so I generated `contrib/msggen/msggen/schema.json` and checked that in as well.

That's what CI is for :)  Good catch.  Changelog is a bit weird, but Release Captain will fix it up when compiling CHANGELOG.md.

Ack  https://github.com/ElementsProject/lightning/commit/dc0ae31950ef20d411fce4418568d9fab744b61a

**Pull request merged**